### PR TITLE
support for providing query params as a dict in responses.add

### PR DESCRIPTION
**Pull request by abehrens**
I wanted to add a mechanism to have the query string built in responses instead of doing so by hand.  I stole some structure from the requests library and tossed in a test and a snippet in the readme.

### CODE REVIEW

**Comment by danihodovic** [comment]
@abehrens are there any future plans on integrating this? It would be a very useful feature.

**Comment by AbdealiJK** [comment]
Came here looking for the same functionality - any plans of adding it ?

File: responses.py [hunk 1]
```diff
@@ -130,14 +131,20 @@ def reset(self):
...
+            content_type='text/plain', json=None, query_params=None):
 
         # if we were passed a `json` argument,
         # override the body and content_type
         if json is not None:
             body = json_module.dumps(json)
             content_type = 'application/json'
 
+        # if we are passed a query_params dict, create the query
+        # string and append to the url
```
Comment (merwok) [inline responses.py:143]: Should the presence of this parameter imply match_querystring=True ?

**Comment by beliaev-maksim** [comment]
@markstory 
I think this may be closed since idea is the same as in #397

**Comment by merwok** [comment]
The other PR does provide a way to do the same thing, although this one was easier to use.

**Comment by beliaev-maksim** [comment]
@merwok 
you can find comments in #397 to see why was decided to go via matchers.

**Pull request closed without merging**